Add basic support for multilingual (#687)
okkur/syna: exampleSite/content/docs/multilingual/content.md

+++
fragment = "content"
weight = 100

title = "MultiLingual Mode"

[sidebar]
  sticky = true
+++

> Syna supports creating Multilingual websites since v0.16.1. The implementation
> only supports `page[.language code].md` method and not 

> > `[language code]/page.md` but we hope to add the second approach in the 

Reviewer: Let's add a tracking issue for the second approach.

> > `[language code]/page.md` but we hope to add the second approach in the 

Reviewer: ```suggestion
> currently supports specifying languages on a per file basis such as `page[.language code].md`
> Using language specific directories `[language code]/page.md` is currently not supported, but might be supported in the future.
```

Reviewer: Great that GH allows multiline comments, but not suggestions :man_facepalming:

> future.

In order to make your website multilingual, you need to change your
configuration so Hugo will generate new pages for translated content in your
project. The way Multilingual mode works is that Hugo generates a new website
for each language and add each page to the website of the same language. For 
example if your website has two languages, English and German, and each page
is translated for both languages, Hugo will generate two websites with 
same pages but in two languages. Hugo will not however generate English pages
for German website if those pages are not translated to German. For more 

> information on Hugo's Multilingual mode, please refer to [their documentation](https://gohugo.io/content-management/multilingual/).

Reviewer: Don't we default to english fragments? We might be doing it differently than hugo here and with that fact the above comment might be confusing. What do you think?

Author: Fragments are used in pages regardless of language. Pages are generated for each language if the page has the translation suffix in their filename. For example `page.md` is default language page and `page.fr.md` is generated for French. However, that is just a semantics thing. Our fragments switch to the language based on Hugo's i18n for not provided text or built-in texts but you can enter any text for any language regardless of page translation.

Reviewer: Ah that makes sense.

> In order to configure two languages, you need to udpate your configuration.

Reviewer: ```suggestion
In order to configure two languages, you need to update your main configuration.
```

> in the future.

Syna allows for fragment overrides in translated pages. This means that when you
translate a page, the translated version will contain all the fragments from the
default language. This allows for fewer duplicated files in your project. This
works the same way as "Translation by filename". For example, let's assume we 
have an `contact` page in our project. The `content` directory might look 
somthing like this:

```
contact/index.md
contact/contact.md
contact/portfolio.md
contact/pricing.md
```

If you want to create a French translation for the `contact` page, and portfolio
fragment inside that page needs to be translated as well, you need to create a
copy from `index.md` and `portfolio.md` and name them `index.fr.md` and
`portfolio.fr.md`, respectively.

```
contact/index.md
contact/index.fr.md
contact/contact.md
contact/portfolio.md
contact/portfolio.fr.md
contact/pricing.md
```

A French version of the page will be generated, containing all your fragments
but the portfolio fragment will be the translated version. The page will have
not changed for the default language.

> > The Multilingual mode is in alpha mode. This means it is subject to change and

Reviewer: ```suggestion
> Multilingual mode is currently release as alpha. It is subject to changes and
```

> may not be stable. Please follow the changelog and updates to this page for

> > more information and news about the changes. Please create issues and pull

Reviewer: ```suggestion
> more information and news about any changes.
> For any issues or contributions please feel free to create issues or pull
```
